Sources/StreamChatUI/Utils/MarkdownFormatter.swift

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public protocol MarkdownFormatter {
+    /// Checks for Markdown patterns in the given String.
+    /// - Parameter text: The string in which Markdown patters are going to be sought.
+    /// - Returns: Returns a Boolean value that indicates whether Markdown patters where found in the given String.
+    func containsMarkdown(text: String) -> Bool
+    /// Returns an attributed string form the given Markdown-formatted string.
+    /// - Parameter string: The string to be formatted.
+    /// - Returns: An attributed string with the corresponding formatted attributes.
+    func format(from string: String) -> NSAttributedString
+}
+
+/// Default implementation for the Markdown formatter
+class DefaultMarkdownFormatter: MarkdownFormatter {
+    enum Attributes {
+        enum Code {
+            static let fontName: String = "CourierNewPSMT"
+            static let color: UIColor = .red
+        }
+    }
+    
+    private let markdownRegex: String =
+        "((?:\\`(.*?)\\`)|(?:\\*{1,2}(.*?)\\*{1,2})|(?:\\~{2}(.*?)\\~{2})|(?:\\_{1,2}(.*?)\\_{1,2})|^(>){1}|(#){1,6}|(=){3,10}|(-){1,3}|(\\d{1,3}\\.)|(?:\\[(.*?)\\])(?:\\((.*?)\\))|(?:\\[(.*?)\\])(?:\\[(.*?)\\])|(\\]\\:))+"
```

Comment:
Yes, I tried simpler approaches with Swift String API but considered that a proper validation of the cases was a good idea and that was better to do with regex. Unfortunately none of the libraries I tested had their own way to check for Markdown syntax public and we do need a way to validate that the String contains regex syntax because we have other situations where text font and color changes depending on if is Jumbo-emoji, is a deleted message, etc. Also I did a quick performance test with the `measure {}`  test you suggested  and for ~1000 times of calling `containsMarkdown()`  (`the measure {`} test does this 10 times ) and validating with the regex, for a fairly long String with several regex elements took around ~0.1s. Let me know what you think.

Reply:
Added screenshots with performance measures on this PR. Please take a look.